test/parallel/test-net-connect-immediate-destroy.js

```diff
@@ -2,7 +2,11 @@
 const common = require('../common');
 const net = require('net');
 
-const socket = net.connect(common.PORT, common.localhostIPv4,
-                           common.mustNotCall());
-socket.on('error', common.mustNotCall());
-socket.destroy();
+const s = net.createServer();
+s.listen(0);
+const port = s.address().port;
+s.close(() => {
+  const socket = net.connect(port, common.localhostIPv4, common.mustNotCall());
+  socket.on('error', common.mustNotCall());
+  socket.destroy();
+});
```

Comment:
I would close the server **after** calling `net.connect()` just to be sure no other process has reclaimed the free port.
```js
'use strict';
const common = require('../common');
const net = require('net');

const s = net.createServer();
s.listen(0);
const port = s.address().port;
const socket = net.connect(port, common.localhostIPv4, common.mustNotCall());
socket.on('error', common.mustNotCall());
s.close();
socket.destroy();
```